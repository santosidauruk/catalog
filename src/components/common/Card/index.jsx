import React, { Component } from 'react'

import { RadioInput, Rate } from '../../common'
import CardImage from './CardImage'
import CardMedia from './CardMedia'
import { CardContent } from './style'

class Card extends Component {
    state = {
        isBuying: false,
    }
    render() {
        const { data, detailed } = this.props
        return (
            <div className="card" style={{ margin: '1rem' }}>
                <CardImage
                    imgSrc={data.images}
                    imgName={data.name}
                    canBeTried={data.can_be_tried}
                    slug={data.slug}
                    hasSlider={detailed}
                />
                <div className="card-content is-paddingless">
                    <CardMedia className="media" data={data} hasDescription={detailed} />
                    {/* If user click 'Beli', show the input */}
                    <CardContent>
                        {this.state.isBuying ? (
                            <div className="notification">
                                <button
                                    className="delete"
                                    onClick={() => this.setState({ isBuying: false })}
                                />
                                <RadioInput
                                    title="Warna yang bisa kamu pilih"
                                    inputs={data.stocks.colors}
                                    name="productColor"
                                />
                                {data.stocks.sizes.length > 0 && (
                                    <RadioInput
                                        title="Size yang bisa kamu pilih"
                                        inputs={data.stocks.sizes}
                                        name="productSize"
                                    />
                                )}
                                <button className="button is-danger">Beli</button>
                            </div>
                        ) : (
                            <React.Fragment>
                                <Rate countOfStars={data.rate} />
                                <button
                                    className="button is-danger"
                                    onClick={() => this.setState({ isBuying: true })}>
                                    Beli
                                </button>
                            </React.Fragment>
                        )}
                    </CardContent>
                </div>
            </div>
        )
    }
}

Card.defaultProps = {
    data: {},
}
export default Card
