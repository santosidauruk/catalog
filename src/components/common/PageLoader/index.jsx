import React from 'react'

import { PageLoaderWrapper } from './style'

const PageLoader = () => (
    <PageLoaderWrapper className="columns is-centered is-flex">
        <span className="icon is-large">
            <i className="fa fa-circle-o-notch fa-pulse fa-2x has-text-danger" />
        </span>
    </PageLoaderWrapper>
)

export default PageLoader
