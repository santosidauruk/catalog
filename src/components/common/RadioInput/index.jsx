import React from 'react'

const RadioInput = ({ inputs, name, title }) => (
    <div className="radio-input">
        <p>{title}:</p>
        <div className="control">
            {inputs.map((item, index) => (
                <label className="radio" key={index}>
                    <input type="radio" name={name} />
                    {item}
                </label>
            ))}
        </div>
    </div>
)

export default RadioInput
