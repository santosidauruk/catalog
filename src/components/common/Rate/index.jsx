import React from 'react'

const Rate = ({ countOfStars }) => (
    <div className="rate">
        {Array.apply(null, { length: 5 })
            .map(Number.call, Number)
            .map((v, index) => (
                <span
                    key={index}
                    className={`fa fa-star ${index + 1 <= countOfStars ? 'rate__checked' : ''}`}
                />
            ))}
    </div>
)

export default Rate
