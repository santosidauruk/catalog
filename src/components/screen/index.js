import ProductList from './ProductList'
import ProductDetail from './ProductDetail'
import NotFound from './NotFound'

export { ProductList, ProductDetail, NotFound }
