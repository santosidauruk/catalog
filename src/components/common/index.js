import Card from './Card'
import PageLoader from './PageLoader'
import RadioInput from './RadioInput'
import Rate from './Rate'

export { Card, RadioInput, PageLoader, Rate }
