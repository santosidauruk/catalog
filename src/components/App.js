import React, { Component } from 'react'
import { Switch, Route } from 'react-router-dom'

import { ProductList } from './screen'
import '../styles/main.css'

class App extends Component {
    render() {
        return (
            <Switch>
                <Route exact path="/" component={ProductList} />
                {/* <Route path="/:productName" component={DetailProduct} /> */}
            </Switch>
        )
    }
}

export default App
