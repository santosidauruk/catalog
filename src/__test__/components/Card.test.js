import React from 'react'
import renderer from 'react-test-renderer'
import BrowserRouter from 'react-router-dom/BrowserRouter'
import { Card } from '../../components/common'

const data = {
    id: 1,
    name: 'Unbranded Wooden Chips',
    slug: 'unbranded-wooden-chips',
    rate: 2,
    price: '9837178',
    images: [
        'https://picsum.photos/420/320?image=31',
        'https://picsum.photos/420/320?image=69',
        'https://picsum.photos/420/320?image=181',
    ],
    can_be_tried: false,
    description:
        'Vitae sint perspiciatis rerum saepe. Id ipsa aperiam expedita sit sint soluta rem aut asperiores. Facere magnam perspiciatis.',
    stocks: {
        colors: ['violet', 'cyan'],
        sizes: ['M', 'XL', 'M'],
    },
}

describe('<Card/> testing', () => {
    it('should rendered correctly', () => {
        const component = renderer.create(
            <BrowserRouter>
                <Card data={data} />
            </BrowserRouter>
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
