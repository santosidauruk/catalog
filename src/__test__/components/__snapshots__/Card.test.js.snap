// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> testing should rendered correctly 1`] = `
<div
  className="card"
  style={
    Object {
      "margin": "1rem",
    }
  }
>
  <div
    className="card-image sc-bwzfXH kKJINq"
  >
    <figure
      className="image is-4by3"
    >
      <img
        alt="Unbranded Wooden Chips"
        src="https://picsum.photos/420/320?image=31"
      />
    </figure>
    <span
      className="icon wish is-medium "
      onClick={[Function]}
    >
      <i
        className="fa fa-heart"
      />
    </span>
    <span
      className="icon zoom"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
    </span>
    <a
      className="card-link"
      href="/unbranded-wooden-chips"
      onClick={[Function]}
    />
  </div>
  <div
    className="card-content is-paddingless"
  >
    <div
      className="media"
    >
      <div
        className="media-content"
      >
        <p
          className="title is-6"
        >
          Unbranded Wooden Chips
        </p>
        <p
          className="subtitle is-6"
        >
          Rp 
          9.837.178
        </p>
      </div>
      <span
        className="tag is-light"
      >
        M, XL, M
      </span>
    </div>
    <div
      className="sc-bdVaJa kGkkgC"
    >
      <div
        className="rate"
      >
        <span
          className="fa fa-star rate__checked"
        />
        <span
          className="fa fa-star rate__checked"
        />
        <span
          className="fa fa-star "
        />
        <span
          className="fa fa-star "
        />
        <span
          className="fa fa-star "
        />
      </div>
      <button
        className="button is-danger"
        onClick={[Function]}
      >
        Beli
      </button>
    </div>
  </div>
</div>
`;
